packages/core/src/oidc/grants/client-credentials.ts

```diff
@@ -61,8 +65,40 @@ export const buildHandler: (
 
   const dPoP = await dpopValidate(ctx);
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  await checkResource(ctx, async () => {});
+  /* === RFC 0001 === */
+  // The value type is `unknown`, which will swallow other type inferences. So we have to cast it
+  // to `Boolean` first.
+  const organizationId = cond(Boolean(params?.organization_id) && String(params?.organization_id));
+  // TODO: Remove
+  if (!EnvSet.values.isDevFeaturesEnabled && organizationId) {
+    throw new InvalidTarget('organization tokens are not supported yet');
+  }
+
+  if (
+    organizationId &&
+    !(await queries.organizations.relations.apps.exists({
+      organizationId,
+      applicationId: client.clientId,
+    }))
+  ) {
+    const error = new AccessDenied('app has not associated with the organization');
+    error.statusCode = 403;
+    throw error;
+  }
+  /* === End RFC 0001 === */
+
+  // Do not check the resource if the organization ID is provided and the resource is not. In this
+  // case, the default resource server will be ignored, and an organization token will be issued.
+  if (!(organizationId && !params?.resource)) {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    await checkResource(ctx, async () => {});
+  }
+
+  const { 0: resourceServer, length } = Object.values(ctx.oidc.resourceServers ?? {});
```

Comment:
Just curious, what is the difference between `params.resource` and `ctx.oidc.resourceServers`? Is `ctx.oidc.resourceServers` the output of the OIDC `checkResource()` method?

Reply:
yes